src/Controls/src/Core/ResourceDictionary.cs

```diff
@@ -47,9 +47,20 @@ public void SetAndLoadSource(Uri value, string resourcePath, Assembly assembly,
 			//this will return a type if the RD as an x:Class element, and codebehind
 			var type = XamlResourceIdAttribute.GetTypeForPath(assembly, resourcePath);
 			if (type != null)
+			{
 				_mergedInstance = s_instances.GetValue(type, (key) => (ResourceDictionary)Activator.CreateInstance(key));
+			}
 			else
+			{
+				if (!FeatureFlags.IsXamlLoadingEnabled)
+				{
+					throw new InvalidOperationException($"The resource '{resourcePath}' has not been compiled using XamlC. "
```

Comment:
This looks like the place which should produce a warning if the feature is enabled. Note that RUC warnings (`IL2026`) have a special behavior and they're not part of the "singlewarn" behavior - so the user should see the actual message, and not just "MAUI has trimming problems". This behavior was specifically designed for cases like this.